src/decisionengine/framework/engine/DecisionEngine.py

```diff
@@ -151,8 +151,11 @@ def rpc_show_de_config(self):
         return self.global_config.dump()
 
     def rpc_print_product(self, product, columns=None, query=None, types=False, format=None):
+        if not isinstance(product, str):
+            raise ValueError(f"Requested product should be a string not {type(product)}")
```

Comment:
I'm expecting strings normally, but could the product be any valid key if someone puts it in the configuration? E.g. a number?

Reply:
As I understand it, the product being checked here is stored by name.  The only valid keys should be strings...

Reply:
@jpunk is correct.  Each product is identified by its name using the (e.g.) `@produces(name1=type1, name2=type2, ...)` syntax, where the keywords are converted to strings.  The check Pat is adding is appropriate.